Unit Tests/Tests/Model/MessageTests.swift

```diff
@@ -0,0 +1,420 @@
+        
+        self.now = Date()
+        
+        let mainContext = self.persistenceController.mainContext
+        
+        mainContext.performAndWait {
+            let firstMessageEntity = mainContext.create(MessageEntity.self)
+            self.firstTestManagedObjectID = firstMessageEntity.objectID
+            
+            let secondMessageEntity = mainContext.create(MessageEntity.self)
+            self.secondTestManagedObjectID = secondMessageEntity.objectID
+            
+            let userEntity = mainContext.create(UserEntity.self)
+            let userEntityObjectID = userEntity.objectID
+            
+            self.firstTestUser = User(identifier: "firstUserIdentifier",
+                                      name: "firstUser",
+                                      avatar: nil,
+                                      presenceState: .unknown,
+                                      metadata: nil,
+                                      createdAt: Date.distantPast,
+                                      updatedAt: self.now,
+                                      objectID: userEntityObjectID)
+            
+            self.secondTestUser = User(identifier: "secondUserIdentifier",
+                                       name: "secondUser",
+                                       avatar: nil,
+                                       presenceState: .unknown,
+                                       metadata: nil,
+                                       createdAt: Date.distantPast,
+                                       updatedAt: self.now,
+                                       objectID: userEntityObjectID)
+            
+            self.thirdTestUser = User(identifier: "thirdUserIdentifier",
+                                      name: "thirdUser",
+                                      avatar: nil,
+                                      presenceState: .unknown,
+                                      metadata: nil,
+                                      createdAt: Date.distantPast,
+                                      updatedAt: self.now,
+                                      objectID: userEntityObjectID)
+        }
+        
+        let testURL = URL(fileURLWithPath: "/dev/null")
+        
+        self.testTextPart = .text("text/plain", "test")
+        self.testLinkPart = .link("image/png", testURL)
+    }
+    
+    // MARK: - Tests
+    
+    func testShouldCreateMessageWithCorrectValues() {
+        let message = Message(identifier: "testIdentifier",
+                              sender: self.firstTestUser,
+                              parts: [self.testTextPart],
+                              readByUsers: [self.firstTestUser, self.secondTestUser, self.thirdTestUser],
+                              lastReadByUsers: [self.secondTestUser],
+                              createdAt: Date.distantPast,
+                              updatedAt: self.now,
+                              deletedAt: Date.distantFuture,
```

Comment:
I don't know if it affects anything, but to be realistic, deleteAt should either be `null`, or in the past. It should never be in the future.

Reply:
It does not affect anything. It allows test to be stable and provide a constant value different from current date.

Reply:
OK. Personally, I'd got for `null` then (if the message is not deleted), as the backend defines deleted to be `deletedAt != null`, not `deleteAt < now`.

Reply:
`testShouldCreateMessageWithCorrectValues()` <- it is hard to test value assignments with `nil` value. None of the values used in unit tests make much sens (intentionally) as that is not what those test are about. Real values should be used in integration tests.